CuratedContent/WorkingEffectivelyWithLegacyCode.md

```diff
@@ -2,23 +2,104 @@
+You may need to add "sensing" variables to help see what you need to see in a unit test.
+
+* **Break dependencies**: Dependencies need to be broken for one of two reasons: Sensing and Separation.
+With **Sensing**, one must be able to inspect the behavior of the code that we can’t otherwise see.  While **Separation** is needed to allow the code to be run in a test harness outside of the production setting.
+Actually breaking the dependencies involves applying minimal refactorings with careful hyper-sensitive editing.
+(There are special dependency-breaking refactorings defined in this book to help with this critical task.)
+
+* **Cover legacy code with unit tests**: If you have the specification for how the targeted legacy code is supposed to work, then write tests to that specification.
+Otherwise, write "Characterization Tests" to see what the code actually does under different input scenarios.
+NOTE: When the official specification differs from the actual observed behavior of the code, go with the actual behavior for because that is what users actually depend on.
+(Differences between specified/desired behavior and actual behavior can be addressed in later iterations if one desires breaking backward compatibility.)
+
+Finally, another major contribution of this book is in detailing the strategies that can be used in the various steps of the Legacy Code Change Algorithm which include:
+
+* **Faking Collaborators:**  Needed to get targeted code into a unit test harness and drive unit tests.:
+  - **Fake Objects**: Impersonates a collaborator to allow sensing and control.
+  - **Mock Objects**: Extended Fake Objects that asserts expected behavior.
+* **Seams**: Ways of inserting test-related code:
+  - **Preprocessing Seams**: Preprocessor macros to replace functions, replace header files, etc.
+  - **Link Seams**: Replace implementation functions (program or system) to define behavior or sense changes.
+  - **Object Seams**: Define interfaces and replace production objects with mock or fake objects in a test harness. (NOTE: Prefer object seams to link or preprocessing seams!)
+
+The Legacy Code Change Algorithm is applied in many small iterations over and over again as pieces of the legacy software are changed.
+Over time, these small incremental refactorings can result in significant improvements to the quality and sustainability of the software and can even perform major (beneficial) architectural changes.
+In fact, at some point, the software may have improved to the point where there is enough strong testing that it is no longer considered legacy code!
+
+This book is packed with practical examples that show nearly every trick there is for refactoring nasty code to break dependencies and getting code into a unit test harness.
+After reading this book, you should be convinced of the need for unit testing and TDD.
+
+<br>
+
+[1]: #ref1 "Fowler, Martin. Refactoring, Addison Wesley, 1999"
```

Comment:
References look good :-)  Thanks for showing us how to do this @markcmiller86!

Reply:
Glad you liked it. I am not sure if you choose to just copy the pattern you saw elsewhere or tried   to follow [this documentation](https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/master/Articles/Blog/ReferencesInMarkdownHybridApproach.md). But, either way, this approach works pretty well when you wanna do lists of references.

Reply:
Nice job!